src/cpp/rtps/transport/TCPTransportInterface.h

```diff
@@ -79,6 +79,9 @@ class TCPTransportInterface : public TransportInterface
     std::vector<fastrtps::rtps::IPFinder::info_IP> current_interfaces_;
     asio::io_service io_service_;
     asio::io_service io_service_timers_;
+    std::shared_ptr<asio::ip::tcp::socket> initial_peer_local_locator_socket_;
+    Locator initial_peer_local_locator_;
```

Comment:
We could just keep the port here (see my other comment regarding `fill_local_physical_port`)